python/python/benchmarks/test_search.py

```diff
@@ -57,6 +69,60 @@ def test_dataset(data_dir: Path) -> lance.LanceDataset:
     return dataset
 
 
+def create_delete_dataset(data_dir):
+    tmp_path = data_dir / "search_dataset_with_delete"
+    dataset = find_or_clean(tmp_path)
+    if dataset:
+        return dataset
+
+    clean_path = data_dir / "search_dataset"
+    shutil.copytree(clean_path, tmp_path)
+
+    dataset = lance.dataset(tmp_path)
+    dataset.delete("filterable % 2 != 0")
```

Comment:
Something to consider for the future: we should split deletion into 2 x 2 matrix with along `contiguous vs random` and `deletion %: 2% vs 20%`.